src/main/java/com/amazonaws/encryptionsdk/CryptoInputStream.java

```diff
@@ -254,4 +257,18 @@ public void setMaxInputLength(long size) {
                 (List<K>) cryptoHandler_.getMasterKeys(),
                 cryptoHandler_.getHeaders());
     }
+
+    AwsCryptoResult<AwsCryptoInputStream> getAwsCryptoResult(AwsCryptoInputStream awsCryptoInputStream) throws IOException {
+        while (!cryptoHandler_.getHeaders().isComplete()) {
+            if (fillOutBytes() == -1) {
+                throw new BadCiphertextException("No CiphertextHeaders found.");
+            }
+        }
```

Comment:
Where was this refactored from? Or is it new?

Reply:
It is copied from CryptoInputStream.getCryptoResult()